library/src/test/java/com/midtrans/java/CoreApiTest.java

```diff
@@ -13,203 +13,189 @@
+        assert result2.has("token_id");
+        assert result2.getString("token_id").matches("[0-9]{6}-[0-9]{4}-[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}");
+        assert result2.getString("status_code").equals("200");
     }
 
     @Test
     public void chargeCreditCardTransaction() throws MidtransError {
+        /* Using method with Global Config */
         dataMockup.setPaymentType("credit_card");
         Map<String, String> cc = new HashMap<>();
-        cc.put("token_id", genCardToken(cardNumberAccept));
+        cc.put("token_id", genCardToken(cardNumberAccept, mainClientKey));
         dataMockup.creditCard(cc);
 
-        JSONObject result = coreApi.chargeTransaction(dataMockup.initDataMock());
-        assert result.length() != 0;
-    }
-
-    @Test
-    public void failChargeWithEmptyBody() throws MidtransError {
-        JSONObject result = coreApi.chargeTransaction(null);
-        assert result.getString("message").equals("An unexpected error occurred");
-    }
+        JSONObject result = CoreApi.chargeTransaction(dataMockup.initDataMock());
+        assert result.get("transaction_status").equals("capture");
 
-    @Test
-    public void chargeGoPayTransaction() throws MidtransError {
-        dataMockup.setPaymentType("gopay");
-        JSONObject result = coreApi.chargeTransaction(dataMockup.initDataMock());
-        assert result.getString("status_code").equals("201");
-        assert result.getString("payment_type").equals("gopay");
-    }
+        /* Using method with config option on request */
+        dataMockup.setPaymentType("credit_card");
+        Map<String, String> cc2 = new HashMap<>();
+        cc2.put("token_id", genCardToken(cardNumberAccept, secondClientKey));
+        dataMockup.creditCard(cc2);
 
-    @Test
-    public void cardPointInquiry() throws MidtransError {
-        JSONObject result = coreApi.cardPointInquiry(genCardToken(cardNumberAccept));
-        assert result.getString("status_code").equals("200");
-        assert result.getString("status_message").equals("Success, Credit Card Point inquiry is successful");
+        JSONObject result2 = CoreApi.chargeTransaction(dataMockup.initDataMock(), configOptions);
+        assert result2.get("transaction_status").equals("capture");
     }
 
     @Test
-    public void orderIdNullCheckTransaction() throws MidtransError {
-        JSONObject result = coreApi.checkTransaction("null");
-        assert result.getString("status_code").equals("404");
-    }
+    public void failChargeWithEmptyBody() {
+        /* Using method with Global Config */
+        try {
+            CoreApi.chargeTransaction(null);
+        } catch (MidtransError e) {
+            e.printStackTrace();
+            assert e.getMessage().contains("Midtrans API is returning API error. HTTP status code: 500 API response: {\"message\":\"An unexpected error occurred\"}");
```

Comment:
Error message from API can change/improved, which if the API change it, this test will break. No need to assert the message value in this general 500 error case.